subprojects/files/src/main/java/org/gradle/internal/file/impl/Deleter.java

```diff
@@ -38,9 +38,17 @@
     boolean deleteRecursively(File target, boolean followSymlinks) throws IOException;
 
     /**
-     * Attempts to clean the given directory recursively, removing all of its contents.
+     * Makes sure that the given target is an empty directory.
+     *
+     * If target is...
+     *
+     * <ul>
+     *     <li>a directory, then its contents are removed recursively,</li>
+     *     <li>a symlink pointing to an existing directory, then the linked directory's contents are removed recursively,</li>
+     *     <li>a file, or a symlink to an existing file, it is deleted and a directory is created in its place,</li>
+     *     <li>non-existent, then a directory is created in its place.</li>
```

Comment:
I don't think the code in `DefaultDeleter` creates a directory if the file is non-existant.

Reply:
You're faster than CI! :) Fixed.